Source/Extensions/Blazorise.DataGrid/_DataGridRowEdit.razor.cs

```diff
@@ -64,6 +66,70 @@ internal protected async Task Save()
         await ParentDataGrid.Save();
     }
 
+    protected async Task HandleCellClick( DataGridColumn<TItem> column )
+    {
+        await ParentDataGrid.HandleCellEdit( column, Item );
+    }
+
+    protected async Task HandleCellKeyDown( KeyboardEventArgs args, DataGridColumn<TItem> column )
+    {
+        if ( ParentDataGrid.IsCellEdit && column.CellEditing )
+        {
+            if ( args.Code == "Escape" )
+            {
+                await Cancel.InvokeAsync();
+                return;
+            }
+
+            if ( args.Code == "Tab" )
+            {
+                await Save();
+
+                if ( ParentDataGrid.EditState == DataGridEditState.Edit )
+                    return;
+
+
+                var currentIdx = OrderedColumnsForEditing.Index( x => x.IsEqual( column ) );
+
+                if ( args.ShiftKey )
+                {
+                    var previousColumn = OrderedColumnsForEditing.ElementAtOrDefault( currentIdx - 1 );
+                    if ( previousColumn is not null )
+                        await ParentDataGrid.HandleCellEdit( previousColumn, Item );
+                    else
+                    {
+                        var currentEditRowIdx = ParentDataGrid.Data.Index( x => x.IsEqual( Item ) );
```

Comment:
DO we need to check for null references here

- `ParentDataGrid.Data?.`
- ParentDataGrid.DisplayData?.
- etc

Reply:
What the heck?
![image](https://github.com/Megabit/Blazorise/assets/22283161/9cfe1246-e575-48e3-a354-9b56e54a0d3b)